dspace-server-webapp/src/test/java/org/dspace/app/rest/AuthorizationRestRepositoryIT.java

```diff
@@ -0,0 +1,1154 @@
+                "invalid-uri",
+                "",
+                "http://localhost/api/wrongcategory/wrongmodel/1",
+                "http://localhost/api/core/sites/this-is-not-an-uuid"
+        };
+        Site site = siteService.findSite(context);
+        SiteRest siteRest = converterService.toRest(site, converterService.getProjection(DefaultProjection.NAME));
+        String siteUri = utils.linkToSingleResource(siteRest, "self").getHref();
+        // disarm the alwaysThrowExceptionFeature
+        configurationService.setProperty("org.dspace.app.rest.authorization.AlwaysThrowExceptionFeature.turnoff", true);
+
+        for (String invalidUri : invalidUris) {
+            System.out.println("Testing the URI: " + invalidUri);
+            // verify that it works for administrators with an invalid or missing uri
+            String adminToken = getAuthToken(admin.getEmail(), password);
+            getClient(adminToken).perform(get("/api/authz/authorizations/search/objectAndFeature")
+                    .param("uri", invalidUri)
+                    .param("feature", alwaysTrue.getName())
+                    .param("eperson", admin.getID().toString()))
+                .andExpect(status().isBadRequest());
+
+            // verify that it works for normal loggedin users with an invalid or missing uri
+            String epersonToken = getAuthToken(eperson.getEmail(), password);
+            getClient(epersonToken).perform(get("/api/authz/authorizations/search/object")
+                    .param("uri", invalidUri)
+                    .param("feature", alwaysTrue.getName())
+                    .param("eperson", eperson.getID().toString()))
+                .andExpect(status().isBadRequest());
+
+            // verify that it works for administators inspecting other users with an invalid or missing uri
+            getClient(adminToken).perform(get("/api/authz/authorizations/search/object")
+                    .param("uri", invalidUri)
+                    .param("feature", alwaysTrue.getName())
+                    .param("eperson", eperson.getID().toString()))
+                .andExpect(status().isBadRequest());
+
+            // verify that it works for anonymous users with an invalid or missing uri
+            getClient().perform(get("/api/authz/authorizations/search/object")
+                    .param("uri", invalidUri)
+                    .param("feature", alwaysTrue.getName()))
+                .andExpect(status().isBadRequest());
+
+            // verify that it works for administrators inspecting anonymous users with an invalid or missing uri
+            getClient(adminToken).perform(get("/api/authz/authorizations/search/object")
+                    .param("uri", invalidUri)
+                    .param("feature", alwaysTrue.getName()))
+                .andExpect(status().isBadRequest());
+        }
+
+        //FIXME add once https://github.com/DSpace/DSpace/pull/2668 is merged
```

Comment:
Again, these tests can be reenabled

Reply:
https://github.com/DSpace/DSpace/pull/2663/commits/81b9a18e81d517db887273eba09b6ca97483b248